### PR TITLE
Implement roles kwarg for guild.create_custom_emoji and emoji.edit

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

**Comment by diceroll123** [comment]
`roles: Optional[list[int]]`

Optional is unnecessary, having Optional means it can be None. You default to an empty list.

File: discord/emoji.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def guild(self):
         name: str
             The new emoji name.
         roles: Optional[list[int]]
```
Comment (Rapptz) [inline discord/emoji.py]: You probably meant ```Optional[List[:class:`Role`]]```

File: discord/emoji.py [hunk 2]
```diff
@@ -243,7 +243,7 @@ def guild(self):
...
-            IDs of the roles for which this emoji is to be whitelisted.  Leave empty to make it unrestricted.
+            List of Roles for which this emoji is to be whitelisted.  Leave empty to make it unrestricted.
```
Comment (Rapptz) [inline discord/emoji.py]: You probably want to reference the `Role` class here, and preface it with "A". I don't like the wording in general.

i.e.

```A :class:`list` or :class:`Roles` that can use this emoji. Leave empty to make it available to everyone.```

File: discord/guild.py [hunk 3]
```diff
@@ -1005,8 +1005,8 @@ def _create_channel(self, name, overwrites, channel_type, category=None, reason=
         image: bytes
             The *bytes-like* object representing the image data to use.
             Only JPG and PNG images are supported.
-        roles: Optional[list[int]]
-            IDs of the roles for which this emoji is to be whitelisted.  Leave empty to make it unrestricted.
+        roles: Optional[list[Role]]
```
Comment (Rapptz) [inline discord/guild.py]: Same comment as the above.

File: discord/guild.py [hunk 4]
```diff
@@ -1005,8 +1005,8 @@ def _create_channel(self, name, overwrites, channel_type, category=None, reason=
...
+            List of Roles for which this emoji is to be whitelisted.  Leave empty to make it unrestricted.
```
Comment (Rapptz) [inline discord/guild.py]: Same nitpick.

**Pull request closed without merging**